### PR TITLE
Feature/mmds metrics

### PR DESCRIPTION
**Pull request by devzbysiu**
# Reason for This PR

Hello :). This PR addresses metrics part of this issue: https://github.com/firecracker-microvm/firecracker/issues/478

## Description of Changes

The changes introduce `mmds_count` and `mmds_fails` metrics for `GET`, `PUT` and `PATCH` requests to MMDS module.
In case of a failure in a parsing logic, `mmds_fails` of particular request type is increased. Each request increases `mmds_count` metric for this request type.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented. - N/A
- [x] Any API changes are reflected in `firecracker/swagger.yaml`. - N/A
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

### CODE REVIEW

File: src/api_server/src/request/mmds.rs [hunk 1]
```diff
@@ -3,37 +3,49 @@
...
+            Err(Error::Generic(
+                StatusCode::BadRequest,
+                format!("Unrecognized PUT request path `{}`.", unrecognized),
+            ))
+        }
     }
 }
 
 pub(crate) fn parse_patch_mmds(body: &Body) -> Result<ParsedRequest, Error> {
     Ok(ParsedRequest::PatchMMDS(
```
Comment (acatangiu) [inline src/api_server/src/request/mmds.rs:44]: Missing `METRICS.patch_api_requests.mmds_count.inc();` here

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 ### Added
 
+- Added MMDS metrics for `GET`, `PUT` and `PATCH` requests
```
Comment (acatangiu) [inline CHANGELOG.md]: I would rephrase to make it clear these are API metrics and not MMDS internal metrics. Maybe smth like:

```
Added API metrics for `GET`, `PUT` and `PATCH` requests on `/mmds` endpoint.
```

Comment (acatangiu) [reply CHANGELOG.md]: Also this commit should be squashed with the previous - updating the changelog can go with the code changes.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,12 +4,12 @@
 
 ### Added
 
-- Added MMDS metrics for `GET`, `PUT` and `PATCH` requests
 - Added devtool build `--ssh-keys` flag to support fetching from private
   git repositories.
 - Added option to configure block device flush.
 - Added `--new-pid-ns` flag to the Jailer in order to spawn the Firecracker
   process in a new PID namespace.
+- Added MMDS metrics for `GET`, `PUT` and `PATCH` requests
```
Comment (acatangiu) [inline CHANGELOG.md:12]: 3rd commit is a _fix_ for 2nd commit, like I said in earlier comment, just squash all commits together.

**Review by acatangiu — COMMENTED** [review]
Changes look good 👍, left a few punctual comments to be addressed.

File: src/api_server/src/request/mmds.rs (hunk 1 above)
Comment (devzbysiu) [reply src/api_server/src/request/mmds.rs:45]: OMG :facepalm: . Sorry for that. Fixed. Tests adjusted as well.

File: CHANGELOG.md (hunk 2 above)
Comment (devzbysiu) [reply CHANGELOG.md]: Done

File: CHANGELOG.md (hunk 3 above)
Comment (devzbysiu) [reply CHANGELOG.md]: Done

**Review by acatangiu — APPROVED** [review]
Looks good!

P.S. don't worry about the snapshot latency perf tests failures - we have an unrelated testing problem we're currently looking at.

**Review by berciuliviu — APPROVED** [review]
LGTM
